### PR TITLE
removed handleDelete in item_list.js.jsx

### PR DESCRIPTION
**Pull request by brendanmcgivern**
@droberts84
- see my comments on item_list.js.jsx

### CODE REVIEW

**Comment by droberts84** [comment]
I'll address the error on the line comment, but one pro tip.. [Google Canary](https://www.google.com/chrome/browser/canary.html), which is GREAT for web development, will display the variable that is undefined instead of the non-helpful error message you see above.  Firefix might also.

File: app/assets/javascripts/components/item_list.js.jsx [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-  handleDelete: function(itemId) {
-    return this.props.onItemDelete(itemId);
-  },
-
   _items: function() {
     var items = this.props.data.map(function (item) {
       return (
-         <Item key={item.id} onItemDelete = {this.handleDelete} item={item} />
+         <Item key={item.id} onItemDelete = {function(itemId) {
+            return this.props.onItemDelete(itemId);
```
Comment (droberts84) [inline app/assets/javascripts/components/item_list.js.jsx]: If put on a `console.log(this);` or `debugger` above this line, you will see that `this` is not equal to what you think it is, causing `this.props` to be undefined.

I meant to talk to you about `this` on Monday before we ran out of time, but it is a hard concept to fully understand (which I still struggle with).  Lets talk tomorrow.

Regardless of `this` problems, IMO moving the `handleDelete` function to an anonymous function in the `Item` call just makes it harder to read and does reduce or simplify your code.

Recall that in JS functions can be assigned to variables, passed around, and then called by the variable they were assigned to.  In this case `this.props.onItemDelete` already contains a function that takes an item id, so your `handleDelete` function wraps it without adding any value.  Instead you could pass the `this.props.onItemDelete` variable directly to your `Item` component, and your `Item` component will call it.

File: app/assets/javascripts/components/item_list.js.jsx [hunk 2]
```diff
@@ -1,13 +1,11 @@
 var ItemList = React.createClass({
```
Comment (droberts84) [inline app/assets/javascripts/components/item_list.js.jsx]: where is your `render` method?

Comment (brendanmcgivern) [inline app/assets/javascripts/components/item_list.js.jsx]: My `render` method is showing up in the branch but it is not showing up in the PR...

```
render: function() {
    var items = this._items();
    return (

      <div className="Table">
        <div className="Heading">
          <div className="Cell table-head">
            <h5>Description</h5>
          </div>
          <div className="Cell table-head">
              <h5>Days Remaining</h5>
          </div>
          <div className="Cell table-head">
              <h5>Complete</h5>
          </div>
        </div>
          {items}
      </div>
      
    );
  }

```

File: app/assets/javascripts/components/item_list.js.jsx (hunk 1 above)
Comment (brendanmcgivern) [inline app/assets/javascripts/components/item_list.js.jsx]: That did it!  I passed in `this.props.onItemDelete` by itself and its working.  I attempted something similar to this but I was passing in `this.props.onItemDelete(itemId)`.  I was getting caught up on the param.

Comment (droberts84) [inline app/assets/javascripts/components/item_list.js.jsx]: Ah, this is an easy mistake to make.  When you pass `
this.props.onItemDelete(itemId)`, the function it refers to is executed
immediately, then the return value is passed to `Item`. `Item` then tries
to call that as a function, but its not a function so you get an error.
 omitting the `()` passes the function itself.

**Pull request merged**